File: app/src/main/java/com/tbm/dogs/activities/viecdangco/HandlerP.java
package com.tbm.dogs.activities.viecdangco;

import android.net.Uri;
import android.os.AsyncTask;
import android.util.Log;

import com.tbm.dogs.Helper.Var;
import com.tbm.dogs.model.obj.Job;

import java.io.IOException;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.ArrayList;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class HandlerP {
    private Results results;
    public HandlerP(Results results){
        this.results = results;
    }

    public void getJobs() {
        new handlerJobs().execute(Var.API_GET_ORDERS,Var.shiper.getHero_id());
    }

    public class handlerJobs extends AsyncTask<String,Void,String>{
        OkHttpClient client = new OkHttpClient();
        @Override
        protected String doInBackground(String... strings) {
            Uri uri = Uri.parse(strings[0])
                    .buildUpon()
                    .appendQueryParameter("hero_id",strings[1])
                    .build();
            Log.e("uriViecdangco",uri.toString());
            URL url = null;
            try {
                url = new URL(uri.toString());
            } catch (MalformedURLException e) {
                e.printStackTrace();
            }
            Request.Builder builder = new Request.Builder();
            builder.url(url);
            builder.addHeader(Var.HEADER,Var.tokenOther);
            Request request = builder.build();
            try {
                Response response = client.newCall(request).execute();
                return response.body().string();
            } catch (IOException e) {
                Log.e("getviecdangco:",e.toString());
            }
            return null;
        }

        @Override
        protected void onPostExecute(String s) {
            super.onPostExecute(s);
            handlerResults(s);
        }

        private void handlerResults(String s) {
            ArrayList<Job> jobs = new ArrayList<>();
            //parse job
            //TODO
        }
    }
}

File: app/src/main/java/com/tbm/dogs/activities/congviec/dangco/HandlerP.kt
package com.tbm.dogs.activities.congviec.dangco

import android.content.Context
import android.util.Log
import com.google.gson.Gson
import com.tbm.dogs.Helper.Maths
import com.tbm.dogs.Helper.Var
import com.tbm.dogs.Helper.VolleyHelper
import com.tbm.dogs.model.obj.Job
import org.json.JSONArray
import org.json.JSONObject

class HandlerP(private val results: Results, val context: Context) {

    private lateinit var maths: Maths
    private var gson: Gson = Gson()
    val volley = VolleyHelper(context)

    fun getJob(tag: String): Job? {
        for (job in Var.jobs!!) {
            if (tag == job.order_id) {
                return job
            }
        }
        return null
    }

    fun getGroupJob(job: Job?): ArrayList<Job> {
        maths = Maths()
        val jobs = ArrayList<Job>()
        for (item in Var.jobs!!) {
            if (maths.distance(job!!.pickup.latitude.toDouble(), job.pickup.longitude.toDouble(), item.pickup.latitude.toDouble(), item.pickup.longitude.toDouble()) < 100) {
                jobs.add(item)
            }
        }
        return jobs
    }

    fun getJobs() {
        handlerJobs(Var.API_GET_NEW_JOBS, Var.shiper?.hero_id!!, "3")
    }

    fun acceptOrder(order_id: String, hero_id: String?, date: String) {
        handlerAcceptOrder(Var.API_ACCEPT_ORDER, order_id, hero_id!!, date)
    }

    private fun handlerAcceptOrder(url: String, order_id: String, hero_id: String, date: String) {
        fun handlerResults(s: String) {
            Log.e("acceptOrder", s)
            val jsonObject = JSONObject(s)
            val status = jsonObject["status"]
            if (status == "success") {
                results.showSuccess(jsonObject["phone_number"] as String)
            } else {
                results.showError()
            }
        }
        volley.requestAPI(
                url,
                HashMap<String, String>().apply {
                    put(Var.HEADER, Var.tokenOther)
                },
                HashMap<String, String>().apply {
                    put("order_id", order_id)
                    put("hero_id", hero_id)
                    put("res_time", date)
                },
                { response -> handlerResults(response) },
                { error ->
                    Log.e("acceptOrder:", error)
                    results.showError()
                }
        )
    }

    private fun handlerJobs(url: String, hero_id: String, service: String) {
        fun handlerResults(s: String) {
            Log.e("getviecdangco", s)
            val jobs = java.util.ArrayList<Job>()
            val jsonObject = JSONObject(s)
            val status = jsonObject["status"]
            if (status == "success") {
                val response = jsonObject["response"] as JSONArray
                if (response.length() > 0) {
                    for (i in 0 until response.length()) {
                        val item = response.getJSONObject(i).toString()
                        val job: Job = gson.fromJson(item, Job::class.java)
                        jobs.add(job)
                    }
                    results.returnJobs(jobs)
                }
            } else {
                results.showError()
            }
        }
        volley.requestAPI(
                url,
                HashMap<String, String>().apply {
                    put(Var.HEADER, Var.tokenOther)
                },
                HashMap<String, String>().apply {
                    put("hero_id", hero_id)
                    put("service", service)
                },
                { response -> handlerResults(response) },
                { error ->
                    Log.e("getviecdangco:", error)
                    results.showError()
                }
        )
    }
}
